src/compositemark/callout.ts

```diff
@@ -0,0 +1,74 @@
+import {Encoding} from '../encoding';
+import {MarkConfig, MarkDef} from '../mark';
+import {GenericUnitSpec, LayerSpec} from '../spec';
+
+
+export const CALLOUT: 'callout' = 'callout';
+export type CALLOUT = typeof CALLOUT;
+
+export interface CalloutDef {
+  type: CALLOUT;
+  calloutOffset: {x: number, y: number};
+  calloutSize: {x: number, y: number};
+  callout: MarkDef;
+  labelOffset:{x: number, y: number};
+  label: MarkDef;
+}
+
+export function isCalloutDef(mark: CALLOUT | CalloutDef): mark is CalloutDef {
+  return !!mark['type'];
+}
+
+export interface CalloutConfig extends MarkConfig {
+  calloutOffset: {x: number, y: number};
+  calloutSize: {x: number, y: number};
+  labelOffset:{x: number, y: number};
+}
+
+export function normalizeCallout(spec: GenericUnitSpec<Encoding<string>, CALLOUT>): LayerSpec {
+  const {mark: mark, selection: _sel, projection: _p, encoding, ...outerSpec} = spec;
+  let calloutOffset: {x: number, y: number} = undefined;
+  let calloutSize: {x: number, y: number} = undefined;
+  let labelOffset: {x: number, y: number} = undefined;
+
+  const defaultOffset = {x: 0, y: 0};
```

Comment:
I have a few issues here

1.  Why should default offset for call out and label be the same?  More importantly, calloutSize shouldn't be zero. 

Instead of inventing offset out of nowhere here.  We should have each of these in `config.callout` with default values already set (just like we have `size` for box -- and read the values from there).